test/integration/smoke/test_deploy_vm_extra_config_data.py

```diff
@@ -0,0 +1,360 @@
+                self.assertEquals(
+                    True,
+                    is_a_match,
+                    'The element from tags from extra config do not match with those found in domain xml'
+                )
+        finally:
+            self.destroy_vm(response.id)
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_update_vm__with_extraconfig_kvm(self):
+        if self.hypervisor.lower() != 'kvm':
+            raise self.skipTest("Skipping test case for non-kvm hypervisor")
+
+        try:
+            # Extra configuration needed to enable hugepages
+            extraconfig = "%3CmemoryBacking%3E%0D%0A++%3Chugepages%2F%3E%0D%0A%3C%2FmemoryBacking%3E"
+
+            response = self.deploy_vm('kvm')
+            vm_id = response.id
+            stop_resp = self.stop_vm(vm_id)
+            update_resp = self.update_vm(vm_id, extraconfig)
+            start_resp = self.start_vm(vm_id)
+
+            host_id = start_resp.hostid
+            host = list_hosts(
+                self.apiclient,
+                id=host_id,
+                hypervisor='kvm')
+
+            instance_name = response.instancename
+            host_ipaddress = host[0].ipaddress
+
+            ssh_client = SshClient(host_ipaddress, port=22,
+                                   user=self.hostConfig['username'],
+                                   passwd=self.hostConfig['password'])
+            virsh_cmd = 'virsh dumpxml %s' % instance_name
+            xml_res = ssh_client.execute(virsh_cmd)
+            xml_as_str = ''.join(xml_res)
+
+            extraconfig_decoded_xml = '<config>' + urllib.unquote(extraconfig) + '</config>'
+
+            # Root XML Elements
+            parser = etree.XMLParser(remove_blank_text=True)
+            domain_xml_root = ET.fromstring(xml_as_str, parser=parser)
+            decoded_xml_root = ET.fromstring(extraconfig_decoded_xml, parser=parser)
+            for child in decoded_xml_root:
+                find_element_in_domain_xml = domain_xml_root.find(child.tag)
+
+                # Fail if extra config is not found in domain xml
+                self.assertNotEquals(
+                    0,
+                    len(find_element_in_domain_xml),
+                    'Element tag from extra config not added to VM'
+                )
+
+                # Compare found XML node with extra config node
+                is_a_match = self.elements_equal(child, find_element_in_domain_xml)
+                self.assertEquals(
+                    True,
+                    is_a_match,
+                    'The element from tags from extra config do not match with those found in domain xml'
+                )
+        finally:
+            self.destroy_vm(response.id)
+
+    @attr(tags=["devcloud", "advanced", "advancedns", "smoke", "basic", "sg"], required_hardware="true")
+    def test_deploy_vm__with_extraconfig_vmware(self):
```

Comment:
This test fails with the following error:
`test_deploy_vm__with_extraconfig_vmware (tests.smoke.test_deploy_vm_extra_config_data.TestAddConfigtoDeployVM): CRITICAL: EXCEPTION: test_deploy_vm__with_extraconfig_vmware: ['Traceback (most recent call last):\n', '  File "/usr/lib64/python2.7/unittest/case.py", line 369, in run\n    testMethod()\n', '  File "/marvin/tests/smoke/test_deploy_vm_extra_config_data.py", line 289, in test_deploy_vm__with_extraconfig_vmware\n    extraconfig_decoded = urllib.unquote(extraconfig)\n', '  File "/usr/lib64/python2.7/urllib.py", line 1217, in unquote\n    bits = s.split(\'%\')\n', "AttributeError: 'NoneType' object has no attribute 'split'\n"]
--------------------- >> end captured logging << ---------------------`

Reply:
Fixed and updated test for vmware

Reply:
===SSH to Host 10.2.2.207 port : 22 SUCCESSFUL===
{Cmd: cat "$(esxcli vm process list | grep i-2-55-VM | tail -1 | awk '{print $3}')" | grep -w 'hypervisor.cpuid.v0 = "FALSE"' via Host: 10.2.2.207} {returns: [u'hypervisor.cpuid.v0 = "FALSE"']}
=== TestName: test_deploy_vm__with_extraconfig_vmware | Status : SUCCESS ===